src/main/java/svc/controllers/CourtController.java

```diff
@@ -30,11 +34,23 @@ CourtsDTO GetCourts() {
 	
 	@ResponseBody
 	@RequestMapping(method = RequestMethod.GET, value="/{id}")
-	Court GetCourt(@PathVariable("id") Long id) {
-		if (id == null) {
-			LogSystem.LogEvent("Null id passed to controller");
-		}
-		
-		return courtManager.GetCourtById(id);
+	Court GetCourt(@PathVariable("id") Long id) throws NotFoundException {
+		Court court = courtManager.GetCourtById(id);
+		if (court == null) 
+			throw new NotFoundException();
+		return court;
 	}
+	
+	@ExceptionHandler(TypeMismatchException.class)
+	@ResponseStatus(value = HttpStatus.BAD_REQUEST, reason="Invalid Court ID")
+	public void typeMismatchExceptionHandler(TypeMismatchException e, HttpServletResponse response){	
+	}
+	
+	@ExceptionHandler(NotFoundException.class)

```

Comment:
This should still be caught in a GlobalExceptionHandler class. The whole point was to not have to re-write this everywhere. I understand the TypeMismatchException scenario but just because we handle that one in the controller doesn't mean we through the global solution out the window...